app/messaging/process-vet-visit.js

```diff
@@ -0,0 +1,48 @@
+const util = require('util')
+const { set } = require('../repositories/vet-visit-repository')
+const { get } = require('../repositories/application-repository')
+const sendMessage = require('../messaging/send-message')
+const { vetVisitResponseMsgType, applicationResponseQueue, serviceUri } = require('../config')
+const { notify: { templateIdVetApplicationComplete, templateIdFarmerApplicationClaim } } = require('../config')
+const sendEmail = require('../lib/send-email')
+
+const processVetVisit = async (message) => {
+  try {
+    const msgBody = message.body
+    console.log('received process vet visit request', util.inspect(msgBody, false, null, true))
+    const { reference } = msgBody.signup
+    const farmerApplication = await get(reference)
+
+    // if no application or application already submitted return
+    if (!farmerApplication || farmerApplication?.vetVisit?.dataValues) {
+      return sendMessage(null, vetVisitResponseMsgType, applicationResponseQueue, { sessionId: msgBody.sessionId })
+    }
+
+    await set({
+      applicationReference: reference,
+      data: JSON.stringify(msgBody),
+      createdBy: 'admin',
+      createdAt: new Date()
+    })
+
+    // Send Vet confirmation email.
+    await sendEmail(
+      templateIdVetApplicationComplete,
+      msgBody.signup.email,
+      { personalisation: { reference }, reference }
+    )
+
+    // Send farmer invitation email.
+    const farmerData = JSON.parse(farmerApplication.data)
```

Comment:
Same as above, can these few lines of code be extracted to a function to convey meaning?